internal/config/read.go

```diff
@@ -69,16 +69,14 @@ func (c *Config) readFromEnv() error {
 
 	parallelism := os.Getenv("BUILDKITE_PARALLEL_JOB_COUNT")
 	parallelismInt, err := strconv.Atoi(parallelism)
-	c.Parallelism = parallelismInt
-	if err != nil {
-		errs.appendFieldError("BUILDKITE_PARALLEL_JOB_COUNT", "was %q, must be a number", parallelism)
+	if err == nil {
```

Comment:
Maybe add a comment why we're ignoring the error, and mention that it will be validated in the `validate` function instead.